core/src/target_os_check.rs

```diff
@@ -0,0 +1,329 @@
+//! Optional checks for `#[cfg(target_os = "target")]
+use crate::parser::get_meta_items;
+use log::{debug, error, log_enabled, warn};
+use quote::ToTokens;
+use syn::{punctuated::Punctuated, Attribute, Expr, ExprLit, Lit, Meta, Token};
+
+#[derive(Copy, Clone, Default, Debug)]
+/// Scoped inside a block that either accepts
+/// or rejects.
+enum TargetScope {
+    #[default]
+    Accept,
+    Reject,
+}
+
+#[derive(Default)]
+/// An iterator that yeilds all meta items and their contained scope.
+struct TargetOsIterator {
+    meta: Vec<(TargetScope, Meta)>,
+}
+
+impl TargetOsIterator {
+    /// Create a new nested meta iterator.
+    fn new(meta: Meta) -> Self {
+        Self {
+            meta: Vec::from([(TargetScope::Accept, meta)]),
+        }
+    }
+}
+
+impl Iterator for TargetOsIterator {
+    type Item = (TargetScope, String);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        while let Some((mut scope, meta)) = self.meta.pop() {
+            if meta.path().is_ident("not") {
+                debug!("encountered not");
+                scope = TargetScope::Reject
+            }
```

Comment:
Should this take into account the existing scope? I.e., if we're already in `TargetScope::Reject` for this node, would `not` turn that into `TargetScope::Accept`? It's super edge-case since I don't know that someone would realistically do this, but how do we handle `#[cfg(not(not(target_os = "ios")))]`?

Reply:
🤔